packages/databyss-services/sources/SourceProvider.js

```diff
@@ -34,8 +41,10 @@ const SourceProvider = ({ children, initialState, reducer }) => {
 
   const getAllSources = () => {
     dispatch(fetchAllSources())
+  }
 
-    //  console.log('here')
+  const getSourcesforPage = id => {
```

Comment:
This needs to use the cache as well.

Reply:
1. Resolve sourceIds for the page
2. Check in cache for some/none/all sourceIds
3. Build list of missing sourceIds
4. Fetch missing sourceIds